library/Core/test/Common/Context.hs

```diff
@@ -1,15 +1,22 @@
 module Common.Context where
 
+import Control.Lens (over)
 import qualified Data.Text as Text
 import qualified Juvix.Core.Common.Context as Context
 import qualified Juvix.Core.Common.NameSpace as NameSpace
 import Juvix.Library
 import qualified Juvix.Library.HashMap as HashMap
+import qualified Juvix.Library.NameSymbol as NameSymbol
+import qualified StmContainers.Map as STM
+import qualified System.IO.Unsafe as Unsafe
 import qualified Test.Tasty as T
 import qualified Test.Tasty.HUnit as T
 
 foo :: Context.T Int Int Int
-foo = Context.empty ("Foo" :| ["Bar", "Baz"])
+foo = Unsafe.unsafePerformIO (Context.empty ("Foo" :| ["Bar", "Baz"]))
+
+unsafeEmpty :: NameSymbol.T -> Context.T term ty sumRep
+unsafeEmpty x = Unsafe.unsafePerformIO (Context.empty x)
```

Comment:
according to the docs, you can't use `atomically` inside the dynamic scope of an `unsafePerformIO`. i just tried it in ghci and got a `NestedAtomically`:

```
[1]» :m Control.Concurrent.STM System.IO.Unsafe 
[2]» let x = unsafePerformIO $ atomically $ newTVar 0
x :: Num a => TVar a
[3]» atomically $ writeTVar x 14
*** Exception: Control.Concurrent.STM.atomically was nested
```

so it's strange that this seems to work?

Reply:
That is odd... Maybe this is differnet if the values are compiled in the file, thus lifting it of some kind of dynamic scoping!?